### PR TITLE
[FX-2192] [WIP] Show header info

### PR DESCRIPTION
**Pull request by dzucconi**
Re: [FX-2192](https://artsyproduct.atlassian.net/browse/FX-2192)

[Figma spec here](https://www.figma.com/file/I5sz0m16SiW7F3tsSoLRxK/Fair-Page?node-id=0%3A1) — in particular note the  "[Web, Desktop Only] Show header, with variable data" section. This broke my brain a little bit.

* First problem: Our existing grid, based on styled-bootstrap-grid, doesn't have gutters/column gaps.
* Second problem: if we were to add column gaps and then set the `noGutter` property on `Col`, this actually doesn't even work how you'd expect it to.
* [Possible solution: a light wrapper around CSS grid](https://github.com/artsy/force/commit/094502d2f57814e09c1ef453c426af0422d31b8d) with the assumption that we have a 12 column grid, with gutters, and assume it collapses into a single column at mobile breakpoints — this could be extracted to Palette as a possible replacement for our current Grid? The question I have is whether or not it's intuitive enough or if there's a better solution we can/should use here.

### CODE REVIEW

**Comment by dzucconi** [comment]
Sigh, meant to open this as a draft PR. But @damassi @zephraph if you can weigh in on the grid, that'd be cool.

File: src/v2/Apps/Show/ShowApp.tsx [hunk 1]
```diff
@@ -24,9 +31,61 @@ export const ShowApp: React.FC<ShowAppProps> = ({ show }) => {
...
+                  </Col>
+
+                  <Col span={1} display={["none", "block"]} />
+
+                  <Col span={5}>
+                    <ShowViewingRoom />
+                  </Col>
+                </>
+              </>
+            )}
```
Comment (dzucconi) [inline src/v2/Apps/Show/ShowApp.tsx:87]: The Figma spec for this broke my brain. There's a different layout for each state of these two booleans.

File: src/v2/Components/FluidGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    return (
+      <__Col__
+        span={[undefined, span]}
+        gridColumn={["1 / -1", undefined]}
+        {...rest}
+      />
+    )
+  }
+
+  return <__Col__ span={span} {...rest} />
```
Comment (dzucconi) [inline src/v2/Components/FluidGrid.tsx:48]: This indirection is a bit weird but basically what I wanted was to simplify the notion of colspan by accepting a number and transforming it to `span N`, however, by default forcing `gridColumn="1 / -1"` for mobile breakpoints which will span the full row.

File: src/v2/Components/FluidGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+/**
+ * Utility for breaking after a non full-width column.
+ * Fills until the end of the row.
+ */
+export const Break = styled(Col)``
+
+Break.defaultProps = {
+  display: ["none", "block"],
+  gridColumn: "auto / -1",
+}
```
Comment (dzucconi) [inline src/v2/Components/FluidGrid.tsx:60]: If anyone has any other ideas about how to "break" a row by force with CSS grid, I'm all ears. This just fills the remainder of the columns — `-1` is the last column of any grid.

File: src/v2/Components/FluidGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    transform: value => value && `span ${value}`,
+  },
+})
+
+type ColSpanProps = { span?: ResponsiveValue<number> } & GridColumnProps
+
+const __Col__ = styled(Box)<ColSpanProps>`
+  ${colSpan}
+  ${gridColumn}
+  outline: 1px dotted red;
```
Comment (dzucconi) [inline src/v2/Components/FluidGrid.tsx:34]: (For debugging)

**Comment by dzucconi** [comment]
@pepopowitz I'm told you may have some opinions here re: grid ❓

**Comment by damassi** [comment]
I wish I knew more about CSSGrid to comment, but yah I would suggest we move this stuff into palette as either another component or an extension to the existing grid.

File: src/v2/Components/FluidGrid.tsx (hunk 2 above)
Comment (dzucconi) [reply src/v2/Components/FluidGrid.tsx:48]: I think I can simplify this a bit and accept valid `gridColumn` values in the `span` shortcut.

**Comment by pepopowitz** [comment]
I think the implementation is great, the example usage provided in this PR makes it look simple & clear to use. I think it'd be a great addition to palette.

**Pull request closed without merging**